float32 x
float32 y
float32[] exits
float32[] exits_raw
float32 current_path_exit
---
float32 exit_chosen